packages/fx-core/src/component/generator/officeAddin/generator.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  FxError,
+  Inputs,
+  Result,
+  ok,
+  err,
+  ManifestUtil,
+  devPreview,
+  ContextV3,
+} from "@microsoft/teamsfx-api";
+import { join } from "path";
+import {
+  AddinNameQuestion,
+  AddinLanguageQuestion,
+  OfficeHostQuestion,
+  getTemplate,
+  AddinProjectFolderQuestion,
+  AddinProjectManifestQuestion,
+} from "./question";
+import { helperMethods } from "./helperMethods";
+import { OfficeAddinManifest } from "office-addin-manifest";
+import projectsJsonData from "./config/projectsJsonData";
+import * as childProcess from "child_process";
+import { promisify } from "util";
+import { CopyFileError } from "../../../core/error";
+import _ from "lodash";
+import { hooks } from "@feathersjs/hooks/lib";
+import { ActionExecutionMW } from "../../middleware/actionExecutionMW";
+import { Generator } from "../generator";
+
+const childProcessExec = promisify(childProcess.exec);
+
+const componentName = "office-addin";
+const telemetryEvent = "generate";
+const templateName = "office-addin";
+
+export class OfficeAddinGenerator {
+  @hooks([
+    ActionExecutionMW({
+      enableTelemetry: true,
+      telemetryComponentName: componentName,
+      telemetryEventName: telemetryEvent,
+      errorSource: componentName,
+    }),
+  ])
+  static async generate(
+    context: ContextV3,
+    inputs: Inputs,
+    destinationPath: string
+  ): Promise<Result<undefined, FxError>> {
+    const result = await OfficeAddinGenerator.doScaffolding(context, inputs, destinationPath);
+    if (result.isErr()) {
+      return err(result.error);
+    }
+
+    const templateRes = await Generator.generateTemplate(
+      context,
+      destinationPath,
+      templateName,
+      "ts"
+    );
+    if (templateRes.isErr()) return err(templateRes.error);
+
+    return ok(undefined);
+  }
+
+  public static async doScaffolding(
+    context: ContextV3,
+    inputs: Inputs,
+    destinationPath: string
+  ): Promise<Result<undefined, FxError>> {
+    const template = getTemplate(inputs);
+    const name = inputs[AddinNameQuestion.name];
+    const addinRoot = destinationPath;
```

Comment:
This is incorrect . . . the addinRoot should be destinationPath + name (make sure this dir also get's created).  The subfolder houses the code for the add-in as this seems to be the pattern for other teams constructions and also allows for future side by side add-ins.

Reply:
The path is changed by design. Because office add-in's taskpane template is self contained and already have a .vscode folder with vscode tasks to run F5. If scaffolded in a subfolder, we need to generate another .vscode in the project root.

Reply:
The project is self-contained because it also supports the legacy workflow that doesn't use the toolkit and is designed to be just an add-in.  My understanding is that the toolkit scenario is supposed to support the combined project structure so that we can have office add-ins and things like launch pages and teams tab in the same project.  If we leave the self-contained structure doesn't that complicate the scenario where a new feature/component get's added to an existing project?

Note that in the old branch I already had code to generate stuff in the .vscode launch.json file in the project directory as well as updating the manifest template.  I still used the manifest in the subfolder for the add-in debugging, but the plan was to update our tooling packages and the launch.json to work with the shared manifest once the toolkit supported deploying it to the new service.

Reply:
OK . . . looks like the other project types are being flattened as well so it looks like that part makes sense (I still get the error).  I guess the how to guides have the responsibility of explaining how to combine different features into a singular project.

Reply:
TEAMSFX_DEBUG_TEMPLATE needs to be turned on to dismiss the error.